src/Facets/PioneerFacet.sol

```diff
@@ -0,0 +1,93 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.17;
+
+import { ILiFi } from "../Interfaces/ILiFi.sol";
+import { LibAsset } from "../Libraries/LibAsset.sol";
+import { LibSwap } from "../Libraries/LibSwap.sol";
+import { ReentrancyGuard } from "../Helpers/ReentrancyGuard.sol";
+import { SwapperV2 } from "../Helpers/SwapperV2.sol";
+import { Validatable } from "../Helpers/Validatable.sol";
+
+/// @title Pioneer Facet
+/// @author LI.FI (https://li.fi)
+/// @notice Main entry point to send bridge requests to Pioneer
+/// @custom:version 1.0.0
+contract PioneerFacet is ILiFi, ReentrancyGuard, SwapperV2, Validatable {
+    /// Storage ///
+
+    bytes32 internal constant NAMESPACE = keccak256("com.lifi.facets.pioneer");
+
+    address payable public immutable PIONEER_ADDRESS;
+
+    /// Errors ///
+
+    error PioneerAddress0();
+
+    /// Constructor ///
+
+    constructor(address payable pioneerAddress) {
+        if (pioneerAddress == address(0)) revert PioneerAddress0();
+
+        PIONEER_ADDRESS = pioneerAddress;
+    }
+
+    /// External Methods ///
+
+    /// @notice Bridges tokens via Pioneer
+    /// @param _bridgeData The core information needed for bridging
+    function startBridgeTokensViaPioneer(
+        ILiFi.BridgeData memory _bridgeData
+    )
+        external
+        payable
+        nonReentrant
+        refundExcessNative(payable(msg.sender))
+        validateBridgeData(_bridgeData)
+        doesNotContainSourceSwaps(_bridgeData)
+        doesNotContainDestinationCalls(_bridgeData)
+    {
+        LibAsset.depositAsset(
+            _bridgeData.sendingAssetId,
+            _bridgeData.minAmount
+        );
+        _startBridge(_bridgeData);
+    }
+
+    /// @notice Performs a swap before bridging via Pioneer
+    /// @param _bridgeData The core information needed for bridging
+    /// @param _swapData An array of swap related data for performing swaps before bridging
+    function swapAndStartBridgeTokensViaPioneer(
+        ILiFi.BridgeData memory _bridgeData,
+        LibSwap.SwapData[] calldata _swapData
+    )
+        external
+        payable
+        nonReentrant
+        refundExcessNative(payable(msg.sender))
+        containsSourceSwaps(_bridgeData)
+        doesNotContainDestinationCalls(_bridgeData)
+        validateBridgeData(_bridgeData)
+    {
+        _bridgeData.minAmount = _depositAndSwap(
+            _bridgeData.transactionId,
+            _bridgeData.minAmount,
+            _swapData,
+            payable(msg.sender)
+        );
+        _startBridge(_bridgeData);
+    }
+
+    /// Internal Methods ///
+
+    /// @dev Contains the business logic for the bridge via Pioneer
+    /// @param _bridgeData The core information needed for bridging
+    function _startBridge(ILiFi.BridgeData memory _bridgeData) internal {
+        LibAsset.transferAsset(
```

Comment:
it's great that you found this helper function but it's a little outdated, we have not been using it for a while as it's not the most gas efficient. We should though. I will take a task to update the library. 

Before we push Pioneer into audit, please could you remind me to ensure that the library has been updated? Otherwise we should adapt the code.

Reply:
FYI: 
https://github.com/lifinance/contracts/pull/1116